[`graphql`]: Remove usage of legacy objects snapshot processing

sergiupopescu199 opened the pull request:
# Description of change

This PR implements changes from upstream, it removes the legacy object snapshot processing in favor of a new approach.

A separate task is solely responsible for updating objects_snapshot, which means the graphql e2e tests, which depend on forcing objects_snapshot changes through an update query, also need to obey this. In lieu of the current behavior, we instead configure the snapshot lag per test scenario, and to get the desired objects_snapshot state

```text
//# advance-clock --duration-ns 1

//# create-checkpoint
```

## Links to any relevant issues

fixes #6470 .

## Type of change

- Enhancement (a non-breaking change which adds functionality)


## How the change has been tested

Made sure that tests passed after the refactoring.

```shell
cargo nextest run -p iota-graphql-e2e-tests --features pg_integration
```

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have checked that new and existing unit tests pass locally with my changes

### Release Notes

<!--
Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates.
-->

- [ ] Protocol:
- [ ] Nodes (Validators and Full nodes):
- [ ] Indexer:
- [ ] JSON-RPC:
- [ ] GraphQL:
- [ ] CLI:
- [ ] Rust SDK:
- [ ] REST API:

sergiupopescu199 committed:
refactor(iota-transactional-test-runner): remove force-object-snapshot-catchup subcommand

sergiupopescu199 committed:
refactor(iota-graphql-rpc): remove force_objects_snapshot_catchup

sergiupopescu199 committed:
refactor(iota-graphql-e2e-tests): remove force-object-snapshot-catchup

Dkwcs reviewed (commented):
I reckon we should remove `force-object-snapshot-catchup` command description/mentions in syntax.md as well.

sergiupopescu199 committed:
update(iota-transactional-test-runner): update syntax.md

sergiupopescu199 committed:
Merge branch 'develop' into sc-platform/issue-6470

sergiupopescu199 commented:
@Dkwcs 

> I reckon we should remove force-object-snapshot-catchup command description/mentions in syntax.md as well.
fixed: https://github.com/iotaledger/iota/pull/6486/commits/bd8a708d5bbadfed2d59b1abb1ff0b379ac690b5.

Dkwcs reviewed (approved):
LGTM

sergiupopescu199 committed:
refactor(iota-indexer): clean up legacy object snapshot codes (#6492)

# Description of change

This PR inherits changes from upstream, it refactors the `iota-indexer`
by removing legacy object snapshot code.

## Links to any relevant issues

fixes #6471 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Ran indexer tests

```shell
cargo test -p iota-indexer --profile simulator --features shared_test_runtime
```

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items
that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have checked that new and existing unit tests pass locally with
my changes

### Release Notes

<!--
Check each box that your changes affect. If none of the boxes relate to
your changes, release notes aren't required.

For each box you select, include information after the relevant heading
that describes the impact of your changes that a user might notice and
any actions they must take to implement updates.
-->

- [ ] Protocol:
- [ ] Nodes (Validators and Full nodes):
- [ ] Indexer:
- [ ] JSON-RPC:
- [ ] GraphQL:
- [ ] CLI:
- [ ] Rust SDK:
- [ ] REST API:

sergiupopescu199 committed:
Merge branch 'develop' into sc-platform/issue-6470
